src/main/java/com/imgix/URLHelper.java

```diff
@@ -177,29 +232,50 @@ public static String encodeURIComponent(String s) {
     return result;
   }
 
+  /**
+   * Accepts a path and encodes it using UTF8 scheme. Also encodes `+$:? ` and decodes `!'()~`. This
+   * method splits the s, or path, on "/" to avoid encoding them. It then encodes each character.
+   * Finally, the method joins the encoded characters on "/" to restore the delimiter.
+   */
   public static String encodeURI(String s) {
-    return s.replaceAll("\\+", "%2B")
-        .replaceAll("\\:", "%3A")
-        .replaceAll("\\?", "%3F")
-        .replaceAll("\\#", "%23")
-        .replaceAll(" ", "%20")
-        .replaceAll("\\%21", "!")
-        .replaceAll("\\%27", "'")
-        .replaceAll("\\%28", "(")
-        .replaceAll("\\%29", ")")
-        .replaceAll("\\%7E", "~");
+    String[] splitString = s.split("/");
+    for (int i = 0; i < splitString.length; i++) {
+      String str = splitString[i];
+      try {
+        splitString[i] =
+            URLEncoder.encode(str, UTF_8)
+                .replaceAll("\\+", "%20")
+                .replaceAll("\\:", "%3A")
+                .replaceAll("\\?", "%3F")
+                .replaceAll("\\#", "%23")
+                .replaceAll(" ", "%20")
+                .replaceAll("\\%21", "!")
+                .replaceAll("\\%27", "'")
+                .replaceAll("\\%28", "(")
+                .replaceAll("\\%29", ")")
+                .replaceAll("\\%7E", "~");
```

Comment:
What's the rationale for decoding these? I imagine it's to ensure consistency but I just wanted to be sure.